### PR TITLE
fix: respect user schema modification by user in warehouse

### PR DESCRIPTION
**Pull request by saurav-malani**
# Description
PR to make sure that the changes done by customer in user schema in warehouse is incorporated in `consolidatedSchema`

## Notion Ticket

https://www.notion.so/rudderstacks/Users-and-Identifies-table-schema-difference-103c3ee744194c9984be4c497527cf74?pvs=4

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: warehouse/schema.go [hunk 1]
```diff
@@ -261,6 +261,12 @@ func overrideUsersWithIdentifiesSchema(consolidatedSchema model.Schema, warehous
 	for k, v := range consolidatedSchema[identifiesTable] {
 		consolidatedSchema[usersTable][k] = v
 	}
+	for k, v := range warehouseSchema[usersTable] {
+		if _, ok := consolidatedSchema[usersTable][k]; !ok {
+			consolidatedSchema[usersTable][k] = v
+			consolidatedSchema[identifiesTable][k] = v
+		}
+	}
```
Comment (achettyiitr) [inline warehouse/schema.go:269]:
```suggestion
	for colName, colType := range warehouseSchema[usersTable] {
		if _, ok := consolidatedSchema[identifiesTable][colName]; !ok {
			consolidatedSchema[identifiesTable][colName] = colType
		}
	}
```

[the remark this answers is not in the document]
Comment (saurav-malani) [reply warehouse/schema.go:269]: Oops, thanks for pointing it out. 😄

File: warehouse/schema.go [hunk 2]
```diff
@@ -237,20 +238,21 @@ func consolidateWarehouseSchema(consolidatedSchema, warehouseSchema model.Schema
...
-func overrideUsersWithIdentifiesSchema(consolidatedSchema model.Schema, warehouseType string) model.Schema {
+func overrideUsersWithIdentifiesSchema(consolidatedSchema model.Schema, warehouseType string, warehouseSchema model.Schema) model.Schema {
 	var (
 		usersTable      = warehouseutils.ToProviderCase(warehouseType, warehouseutils.UsersTable)
 		identifiesTable = warehouseutils.ToProviderCase(warehouseType, warehouseutils.IdentifiesTable)
 		userIDColumn    = warehouseutils.ToProviderCase(warehouseType, "user_id")
 		IDColumn        = warehouseutils.ToProviderCase(warehouseType, "id")
 	)
-
+	fmt.Println("consolidatedSchema before override: ", consolidatedSchema[identifiesTable])
```
Comment (lvrach) [inline warehouse/schema.go:255]:
```suggestion
```

File: warehouse/schema.go [hunk 3]
```diff
@@ -261,10 +263,15 @@ func overrideUsersWithIdentifiesSchema(consolidatedSchema model.Schema, warehous
...
+	for k, v := range warehouseSchema[usersTable] {
+		if _, ok := warehouseSchema[identifiesTable][k]; !ok {
+			consolidatedSchema[usersTable][k] = v
+			consolidatedSchema[identifiesTable][k] = v
+		}
+	}
 	consolidatedSchema[usersTable][IDColumn] = consolidatedSchema[identifiesTable][userIDColumn]
 	delete(consolidatedSchema[usersTable], userIDColumn)
-
+	fmt.Println("consolidatedSchema after override: ", consolidatedSchema[identifiesTable])
```
Comment (lvrach) [inline warehouse/schema.go:274]:
```suggestion
```

**Review by lvrach — CHANGES_REQUESTED** [review]

**Review by achettyiitr — APPROVED** [review]

File: warehouse/schema.go (hunk 2 above)
Comment (saurav-malani) [reply warehouse/schema.go]: sorry, removed.

**Review by lvrach — APPROVED** [review]

**Pull request merged**